src/core/presentation_submission.rs

```diff
@@ -0,0 +1,151 @@
+}
+
+impl DescriptorMap {
+    /// The descriptor map MUST include an `id` property. The value of this property MUST be a string that matches the `id` property of the [InputDescriptor::id()] in the [PresentationDefinition] that this [PresentationSubmission] is related to.
+    ///
+    /// The descriptor map object MUST include a `format` property. The value of this property MUST be a string that matches one of the [ClaimFormatDesignation]. This denotes the data format of the [Claim](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:claim).
+    ///
+    /// The descriptor map object MUST include a `path` property. The value of this property MUST be a [JSONPath](https://goessner.net/articles/JsonPath/) string expression. The path property indicates the [Claim](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:claim) submitted in relation to the identified [InputDescriptor], when executed against the top-level of the object the [PresentationSubmission] is embedded within.
+    ///
+    /// For more information, see: [https://identity.foundation/presentation-exchange/spec/v2.0.0/#presentation-submission](https://identity.foundation/presentation-exchange/spec/v2.0.0/#presentation-submission)
+    pub fn new(
+        id: impl Into<DescriptorMapId>,
+        format: ClaimFormatDesignation,
+        path: JsonPath,
+    ) -> Self {
+        Self {
+            id: id.into(),
+            format,
+            path,
+            path_nested: None,
+        }
+    }
+
+    /// Return the id of the descriptor map.
+    pub fn id(&self) -> &DescriptorMapId {
+        &self.id
+    }
+
+    /// Return the format of the descriptor map.
+    ///
+    /// The value of this property MUST be a string that matches one of the
+    /// [ClaimFormatDesignation]. This denotes the data format of the Claim.
+    ///
+    /// See: [https://identity.foundation/presentation-exchange/spec/v2.0.0/#presentation-submission](https://identity.foundation/presentation-exchange/spec/v2.0.0/#presentation-submission)
+    pub fn format(&self) -> &ClaimFormatDesignation {
+        &self.format
+    }
+
+    /// Return the path of the descriptor map.
+    pub fn path(&self) -> &JsonPath {
+        &self.path
+    }
+
+    /// Set the nested path of the descriptor map.
+    ///
+    /// The format of a path_nested object mirrors that of a [DescriptorMap] property. The nesting may be any number of levels deep.
+    /// The `id` property MUST be the same for each level of nesting.
+    ///
+    /// > The path property inside each `path_nested` property provides a relative path within a given nested value.
+    ///
+    /// See: [https://identity.foundation/presentation-exchange/spec/v2.0.0/#processing-of-submission-entries](https://identity.foundation/presentation-exchange/spec/v2.0.0/#processing-of-submission-entries)
+    ///
+    /// Errors:
+    /// - The id of the nested path must be the same as the parent id.
```

Comment:
TODO: Remove this error message, we're no longer throwing an error in this method.